Add RFC to discuss RustConf 2024 Steering Committee (#3549)
rust-lang/rfcs: text/rustconf-steering-committee.md

- Feature Name: rustconf-steering-committee
- Start Date: 12-27-2023
- RFC PR: 
- Rust Issue: N/A

# Summary

This RFC proposes creating a RustConf steering committee to represent the Rust Project’s perspectives and priorities throughout the RustConf 2024 planning process. 

# Proposal

The Rust Foundation staff team would like to propose that a RustConf steering committee composed of ~5 Rust Project members be formed with an agreed-upon process of member appointment and scope established.

A Rust Project steering committee for RustConf could be defined as:

> A group of ~5 volunteers from within the Leadership Council and the Project Directors responsible for providing input and direction to the Rust Foundation staff team regarding RustConf 2024. The Rust Foundation will lean on the steering committee to advocate for areas of desired involvement besides the traditional UnConference Day and Program Committee. 

Reviewer: > from within the Leadership Council and the Project Directors

it's probably useful if those people are involved too, but should the group be limited like this? it might limit people with prior conf organizing experience from participating and probably also increases the work load of the council members and project directors.

Reviewer 2: I agree here. In my opinion, it would make sense to have 1-2 PDs, 1-2 council reps, and 0-2 others.

Reviewer 3: Taking a step back, the most important point is that the program committee reflect a cross cut of the Rust Project and the wider Rust community. We have a mechanism in the Project for reflecting "Project will" that does not require polling all members of the Project: the Council. Unfortunately, we don't have such a mechanism for the wider community, and it seems like it would be an extremely large amount of work in and of itself to attempt to determine such a mechanism. 

With all of that said, I think this RFC should not layout who forms the Steering Committee but rather who _selects_ the Steering Committee membership and what the criteria for that selection are. This allows for flexibility to pick Steering Committee members who fulfill the prescribed criteria without artificially limiting the pool of potential candidates.

## Straw-man Proposal

My straw-man proposal would thus be the following:

### Who selects the steering committee?

The Rust Project Leadership Council is responsible for choosing the steering committee based on the criteria below.

### Criteria for the Committee as a whole
* 6-9 members total
* No more than 2 members may share the same affiliation (using the definition of affiliation used by the Council already)
* A simple majority must be Project members
* TODO: ideally we'd have some criteria for demographic diversity but putting such criteria into strict quantatative terms can be difficult.

### Criteria for individual members
* Must have the time and energy to engage in the process
* Must not be currently under moderation team sanction
* Must have some qualification that speaks towards their knowledge of Rust and the Rust community including but not limited to Project membership, authorship of well known ecosystem crate(s), prominent Rust author/speaker/blogger/etc., organizer of another Rust conference, etc.

Once a steering committee has been formed, we can establish a regular process of communication between the Rust Foundation, the steering committee, and other groups from within the Project as identified by the steering committee. 

# Motivation

While the Rust Foundation has served as a Diamond-level sponsor of RustConf since our inception in 2021, we have sought approval to use our staff and financial resources towards planning and managing RustConf in a larger way in 2024. We have gotten approval from our board and received [encouragement](https://rust-lang.zulipchat.com/#narrow/stream/335408-foundation/topic/RustConf.202024.20next.20steps/near/403266679) from the Project and Leadership Council to proceed in Q4.

> Although we have always intended for the Program Committee to remain independent (with management and facilitation provided by the Foundation), Ryan Levick commented that additional Project input throughout the planning process is desired. The staff team discussed this internally and decided that a steering committee would be the right vehicle to identify and drive the type of Rust Project input requested. 

Reviewer: ```suggestion
Although we have always intended for the Program Committee to remain independent (with management and facilitation provided by the Foundation), Ryan Levick commented that additional Project input throughout the planning process is also desired. The staff team discussed this internally and decided that a steering committee would be the right vehicle to identify and drive the type of Rust Project input requested. 
```

When the Rust Foundation staff team raised this idea to our Project Directors during our biweekly meeting, we were asked to outline our idea in an RFC.   

# Practicalities

Once this proposal is accepted, the Rust Foundation team can work with the Leadership Council and the Project Directors to carry out a volunteering process to form the steering committee. 

> Once the steering committee is staffed, we can agree upon a channel and cadence of communication. From there, we will be asking the steering committee to share a clear set of expectations about their preferred involvement in RustConf planning (if there are requests outside of Program Committee and UnConf involvement). We can discuss these desires and expectations in Zulip.

Reviewer: > a clear set of expectations about their preferred involvement in RustConf planning

it might be good to give a rough list of what areas of planning exist when organizing such a conference. that would make it a lot easier for the volunteer team to figure out in which areas they would like to be involved. the list would not need to be part of the RFC, but it would probably be good to have during the volunteering process, so that people know what they would sign up for. :)

# Summary

> This RFC proposes creating an advisory Rust Steering Committee for RustConf to represent Rust Project and community perspectives and priorities throughout the RustConf 2024 planning process. 

Reviewer 3: If this RFC is purposefully limited just to the 2024 RustConf than I'm not sure an RFC is appropriate. 

My initial impression of what this document would be was a document for establishing how RustConf would be run at a high-level. Such a document would clearly layout who has authority over which decisions. For example, I imagined that the RFC would have said something like: 

## Areas of Responsibility

The following groups are responsible for the following areas. More detail on the nature and limits of the responsibility can be found below:
* The Rust Foundation Staff: logistical operation of the event itself, coordinating sponsor relations, ...
* The Steering Committee: all programming aspects of the event (e.g., program committee, MC selection, format of talks, whether to include workshops, inclusion of an UnConf day(s) or other supplementary events, etc.)

# Definitions

- "Steering Committee": A group of advisors that present recommendations to an organizing group (in this case, the "organizing group" being the Rust Foundation as the organizer of RustConf 2024). Often, a steering committee will be in place to represent the interests of a specific group (in this case, the Rust Project and community)
- "Program Committee": The group responsible for selecting speakers/talks to appear on the RustConf agenda/"program". The Program Committee for RustConf has historically been independent of the organizer and composed largely of members of the Rust Project. The Foundation intends to maintain this tradition in 2024 but will provide oversight and facilitation.
- "CFP": Stands for "Call for Proposals". This is the period and process during which we will solicit talk submissions for RustConf. The Program Committee will select submissions from the CFP to appear on the program and might also advocate for inviting several speakers who did not submit a talk. 
  
# Proposal

The Rust Foundation staff team is proposing the formation of a RustConf Steering Committee with the following parameters:

> - Members: 4 total members from the Rust Project -- one leadership council member, one Project Director member, and two members from the wider Rust community. We feel that an even split between leadership members and community members makes sense as the former positions can provide context and Project authority while the latter can advocate for those who are not necessarily in a position of power within the Project. We will lean on the Leadership Council as a whole to select this group based on a process of self-nomination initiated by the Foundation and promoted by the Leadership Council.  

Reviewer 3: The split between Project and wider community makes sense to me, but I'm unsure why we're being so prescriptive about who represents the Project. What is the advantage of this over 2 members who represent the Project as selected by the Leadership Council? Requiring that the members be Leadership Council members or Project Directors seems like a good way to artificially limit the pool of who can participate without much benefit. 

---
> We will lean on the Leadership Council as a whole to select this group based on a process of self-nomination initiated by the Foundation and promoted by the Leadership Council.

Nit: the way this is worded seems odd to me. I think we can phrase it much directly:

> The Leadership Council is responsible for establishing the process by which the members of the Steering Committee are selected. 

With that being said, are we sure we really want to leave it up to the Leadership Council to establish their own rules which are not subject to wider review? It might make sense for this RFC to lay out some base rules

Reviewer: how about this?

```suggestion
- Members: 4 total members from the Rust Project -- e.g. one leadership council member, one Project Director member, and two members from the wider Rust community. The Leadership Council is responsible for establishing the process by which the members of the Steering Committee are selected.
```

Reviewer 3: That's fine with me - my only nit is that "wider Rust community" would indicate to me people who are explicitly not members of the Rust Project which sort of contradicts what you stated previously.

> - Scope: The Foundation would like the Steering Committee's involvement in the following areas:

Reviewer 3: Nit: I doesn't strike me as normal for an RFC to talk about the desires of the author unless that information is relevant for justifying why the authors have chosen a certain way between tradeoffs or if the authors would like to throw in a "nice-to-have" loose requirement. For more foundational information, a more direct wording is preferred: 

```suggestion
- Scope: The Steering Committee's involvement will be in the following areas:
```

>  > - **Program Committee:** The Steering Committee should establish the structure, personnel, and processes of the Program Committee with oversight and collaboration from the Rust Foundation. The Foundation will take this plan from the Steering Committee to formally set up the Program Committee which will select talks after the closure of the call for talk proposals. Steering Committee members will not be on the Program Committee or be involved in selecting talks, although they can advise the Program Committee on key priorities as talks are being selected. To stay on schedule, the Foundation will likely need to open the CFP earlier than these two committees are in place, however, we will share our intentions for talk tracks/suggested topics with the Leadership Council in advance for feedback. 

Reviewer 3: > with oversight and collaboration from the Rust Foundation

What are the limits to this oversight and collaboration. Is the Foundation allowed to disband the Steering Committee? Is the Steering Committee making proposals that the Foundation Staff ultimately has the right to give their approval or disapproval of?

> >  - **UnConf Day:** The Foundation would be pleased to offer an UnConf Day again in 2024, but would lean on the Steering Committee to decide this. If they decide there should be an UnConf Day after RustConf, the Foundation would like the Steering Committee to provide us with a basic plan including content and identification of people from the Project we should be logistically coordinating with ("UnConf Day Organizers") to ensure proper supplies and resources are available to UnConf organizers and attendees. The Foundation would welcome the Steering Committee's participation in planning and running the UnConf alongside the Organizers they identify.

Reviewer 3: Again, this is written too much from the author's perspective. Try to avoid an us/them framing and talk about all groups in the third person. 
```suggestion
>  - **UnConf Day:** The Steering Committee is in charge of deciding whether to include an Unconf style event in the programming. If they decide there should be an UnConf Day after RustConf, the Steering Committee is required to provide the Foundation with a basic plan including content and identification of people from the Project that should be involved in coordinating logistics.
```

> > - **Other considerations:** The Foundation would also like to lean on the Steering Committee to understand if there are any other areas of the event they would like to be involved in on behalf of the Rust Project. The Rust Foundation needs to operate independently on major event decisions (like co-located events and conference activities, tickets/access, general event experience, and sponsorship experience) but will welcome feedback from the Steering Committee along the way.

Reviewer 3: > The Foundation would also like to lean on the Steering Committee to understand if there are any other areas of the event they would like to be involved in on behalf of the Rust Project.

This seems wishy-washy in a way that is unhelpful. I think what you're trying to say is that the various groups should make a best effort to involve input from other groups in their areas of responsibility. I think we should state that.

> All responsible groups should make an active effort to involve the input of other groups in their areas of responsibility. For example, the Foundation Staff should coordinate with the steering committee on event location before making their ultimate decision. Likewise, the Steering Committee should involve Foundation Staff in discussions on supplementary events even before they make a final decisions to minimize difficulties with logistics planning. 

> he Rust Foundation needs to operate independently on major event decisions (like co-located events and conference activities, tickets/access, general event experience, and sponsorship experience)

You've stated this mater of factly, but I'm not sure I agree that the Staff *must* be able to act independently in all of these areas in order to effectively organize RustConf. While logisical concerns like Sponsorship experience and tickets/access seem to fall squarely in the "operations" bucket (and thus appropriate for the Staff to be charged with executing on them more or less independently), "co-located events", "conference activities", and "general event experience" seem to be much more of a programming concern, and thus I would have imagined they would have falled squarely in the responsibility of the Steering Committee.

- **Communication:** The Rust Foundation will share monthly asynchronous updates with the Steering Committee as the RustConf planning process develops. These updates will be used to share advanced notice of key event milestones (i.e. CFP open/close date, registration details, major planning changes/developments). The steering committee can use an agreed-upon communication channel to ask questions/provide feedback about the details shared. 

# Motivation

> Since it was approved by the board and leadership council for the Rust Foundation to play a larger role in managing RustConf, we have intended to facilitate an UnConf Day and for the Program Committee to remain independent (with management and facilitation provided by the Foundation). We also understand that there may be additional expectations around Rust Project involvement and feel that clearer parameters around the Rust Project's role in RustConf would be helpful.

Reviewer 3: ```suggestion
Since it was approved by the Rust Foundation Board and leadership council for the Rust Foundation to play a larger role in managing RustConf, we have intended to facilitate an UnConf Day and for the Program Committee to remain independent (with management and facilitation provided by the Foundation). We also understand that there may be additional expectations around Rust Project involvement and feel that clearer parameters around the Rust Project's role in RustConf would be helpful.
```

The Foundation's senior management team discussed this internally and decided that a Rust Project Steering Committee for RustConf would be the right vehicle to identify and drive further Rust Project input. We feel this is the right approach due to the nature of event planning -- many decisions need to be made quickly to deliver a high-quality event for the community. We will be working with many, many vendors, sponsors, and stakeholders and need as clear a project management process as possible. An open thread on Zulip, a closed thread on Zulip without clarity on deadlines or decision-makers, or an ill-defined process of collaboration will not enable the Foundation to plan a satisfactory event. 

> When the Rust Foundation staff team raised the idea of a Steering Committee to our Project Directors during our biweekly meeting, we were asked to outline our idea in an RFC.   

Reviewer 3: I don't think this is really necessary to say personally.
```suggestion
```
